src/modules/RA/DataLayer/components/SourceFetcher.js

```diff
@@ -23,12 +23,19 @@ const SourceFetcher = ({ dispatch, dataProvider, sourceId, fields = [] }) => {
 
     const fillFields = async () => {
       const { data: { _type: type, fields: sourceFields = [] } } = await load(sourceId);
-      const filledFields = sourceFields.map(({ id, name, label }) => ({
-        id,
-        name,
-        label,
-        ...fields.find(({ id: fieldId }) => id === fieldId) || {},
+      const fieldsFromSource = sourceFields.filter(
+        ({ id }) => !fields.find(({ id: fieldId }) => id === fieldId),
```

Comment:
Using [`.some()`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/some) instead of `.find()` would return directly a boolean.